src/services/download-service.ts

```diff
@@ -12,123 +13,153 @@ class DownloadService {
     return !(this.player.isLive() || this.player.getVideoElement().mediaKeys);
   }
 
-  private isContentTypeSupported(response: Response) {
-    if (response.ok) {
-      const contentType = response.headers.get('content-type');
-      return !!(contentType && contentType.toLowerCase().includes('video'));
-    }
-    return false;
-  }
-
-  private getFilename(response: Response) {
-    if (this.player.isImage()) {
-      return this.player.sources?.metadata?.name || 'image';
-    }
-    const responseUrlSplit = response.url.split('/');
+  private getFilename(metadata: DownloadMetadata) {
+    if (this.player.sources.metadata?.name) return this.player.sources.metadata?.name;
+    if (this.player.isImage()) return 'image';
+    const responseUrlSplit = metadata!.flavors[0].downloadUrl.split('/');
     return responseUrlSplit[responseUrlSplit.indexOf('fileName') + 1];
```

Comment:
is it guaranteed to contain "fileName" ?

Reply:
@lianbenjamin

Reply:
@SivanA-Kaltura yes.. basically i kept the old code for VODs and just added the part for image.